pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go

```diff
@@ -1878,3 +1878,45 @@ func Test_generateObjectMeta(t *testing.T) {
 		})
 	}
 }
+
+func TestReconcileCleanupCheck(t *testing.T) {
+	elWithCustomResourceForNodeSelector := makeEL(withStatus, withKnativeStatus, func(el *v1alpha1.EventListener) {
+		el.Spec.Resources.CustomResource = &v1alpha1.CustomResource{
+			RawExtension: test.RawExtension(t, duckv1.WithPod{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Service",
+					APIVersion: "serving.knative.dev/v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: generatedResourceName,
+				},
+				Spec: duckv1.WithPodSpec{
+					Template: duckv1.PodSpecable{
+						Spec: corev1.PodSpec{
+							NodeSelector: map[string]string{
+								"hi": "hello",
+							},
+						},
+					}},
+			}),
+		}
+	})
+	startResources := test.Resources{
+		EventListeners: []*v1alpha1.EventListener{elWithCustomResourceForNodeSelector},
```

Comment:
making sure I'm understanding this test properly:

the initial state is that we have a k8s deployment and an EL with customResource. We should expect the end state to be that we have a knative service present but not the deployment. Right? If so, shouldn't we look at the resources remaining after reconcile and verify that we do not have a deployment/service present?

Reply:
Yes you are right

I have updated testcase to validate the resources left after reconcile